app/wasm/queries.go

```diff
@@ -26,6 +27,26 @@ func NewQueryPlugin(
 	}
 }
 
+func (qp QueryPlugin) GetFullDenom(ctx sdk.Context, contract string, subDenom string) (*string, error) {
+	// Address validation
+	contractAddress, err := sdk.AccAddressFromBech32(contract)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "address from bech32")
+	}
+	err = sdk.VerifyAddressFormat(contractAddress)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "verify address format")
+	}
+	// TODO: sub-denom validations
```

Comment:
Can you do the validation as part of the PR?

Ideally as a separate function `ValidateSubDenom(string) error`

Reply:
I will during the afternoon. Will scour the osmosis docs to see what the requirements for sub-denom are.